server/data/data-cassandra/src/main/java/org/apache/james/user/cassandra/CassandraRepositoryConfiguration.java

```diff
@@ -0,0 +1,56 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.user.cassandra;
+
+import org.apache.commons.configuration2.HierarchicalConfiguration;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.commons.configuration2.tree.ImmutableNode;
+import org.apache.james.user.lib.model.Algorithm;
+
+public class CassandraRepositoryConfiguration {
+    public static final String DEFAULT_ALGORITHM = "SHA-512/salted";
+    public static final String DEFAULT_HASHING_MODE = Algorithm.HashingMode.PLAIN.name();
+
+    public static final CassandraRepositoryConfiguration DEFAULT = new CassandraRepositoryConfiguration(
+        Algorithm.of(DEFAULT_ALGORITHM), DEFAULT_HASHING_MODE
+    );
+
+    private final Algorithm preferredAlgorithm;
+    private final String fallbackHashingMode;
```

Comment:
Maybe this deserves a stronger type (`HashingMode`?)

Reply:
Right, since we have that type, why not use it :-)